### ISSUE TITLE
Accessibility Issue: Multiple Links on Same Line in Flutter RichText

### ISSUE THREAD

**Issue by Nishorecolan** [flutter/flutter#159127]
### Steps to reproduce

1. - Create a RichText widget with multiple TextSpan or WidgetSpan elements, each representing a clickable link.
2. - Wrap each link in a Semantics widget to ensure accessibility support.
3. - Add GestureDetector to handle onTap events for each link.
4. - Test using a screen reader (TalkBack on Android or VoiceOver on iOS).

### Expected results

Each link (e.g., "Dart" and "widgets") should be announced as a distinct link by the screen reader.

Both links should be focusable and clickable independently.

### Actual results

The screen reader only selects or announces the first link consistently.

The second link is often skipped or merged with surrounding text, making it inaccessible.

### Code sample

class RichTextWithLinks extends StatelessWidget {

  const RichTextWithLinks({super.key});

  @override

  Widget build(BuildContext context) {

    return MaterialApp(

      home: Scaffold(

        appBar: AppBar(

          title: const Text('Rich Text Links'),

        ),

        body: Padding(

          padding: const EdgeInsets.all(16.0),

          child: MergeSemantics(

            child: RichText(
  

              text: TextSpan(

                style: const TextStyle(color: Colors.black, fontSize: 16.0),

                children: [

                  const TextSpan(

                    text:

                        'Android is a mobile operating system based on a modified version of the ',

                  ),

                  WidgetSpan(
  alignment: PlaceholderAlignment.baseline,
                  baseline: TextBaseline.alphabetic,

                    child: Semantics(

                      label: 'Linux kernel',

                      button: true,

                      //link: true,

                      child: GestureDetector(

                        onTap: () {

                          print("Link 1 tapped!");

                        },

                        child: const Text(

                          'Linux kernel',

                          style: TextStyle(

                            color: Colors.blue,

                            decoration: TextDecoration.underline,

                          ),

                        ),

                      ),

                    ),

                  ),

                  const TextSpan(

                    text: ' and other ',

                  ),

                  WidgetSpan(
  alignment: PlaceholderAlignment.baseline,
                  baseline: TextBaseline.alphabetic,

                    child: Semantics(

                      label: 'open-source',

                      button: true,

                      //link: true,

                      child: GestureDetector(

                        onTap: () {

                          print(" link 2 tapped!");

                        },

                        child: const Text(

                          'open-source',

                          style: TextStyle(

                            color: Colors.blue,

                            decoration: TextDecoration.underline,

                          ),

                        ),

                      ),

                    ),

                  ),

                  const TextSpan(

                    text: ' software.',

                  ),

                ],

              ),

            ),

          ),

        ),

      ),

    );

  }

}

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by huycozy** [comment]
Hi @Nishorecolan 
Can you share the reason why you use MergeSemantics here? If I remove `MergeSemantics`, links are clickable independently as expected.

**Comment by Nishorecolan** [comment]
My rich text content should be read in a single accessibility tap and links should be clickable independently.

**Comment by huycozy** [comment]
In case of using MergeSemantics, I see that URLs (WidgetSpan) are read last, after the reader has read all the TextSpan(s), and it is similar to another issue at https://github.com/flutter/flutter/issues/51804.

Reproduced this issue on Flutter stable `3.24.5` and master `3.27.0-1.0.pre.552`. The issue appears on both Android TalkBack and iOS VoiceOver readers.

**Comment by Nishore-Dev** [comment]
@huycozy, Do we have any workaround to fix this?

**Comment by huycozy** [comment]
I don't see a workaround now. Please wait for insight from accessibility team.

**Comment by Nishore-Dev** [comment]
Okay. Thanks for your update.

**Comment by chunhtai** [comment]
I don't think there is an easy way to do as of now, you may be able to get around by converting everything into widget span since there isn't a way to concatenate text from widget span in between textspan.
